### PR TITLE
Gradient flow debug, take type and variational from parameter file

### PR DESCRIPTION
**Pull request by fedetask**

### CODE REVIEW

File: utilities/debug_utils.py [hunk 1]
```diff
@@ -0,0 +1,126 @@
+import yaml
+import sys
+import os
+
+from matplotlib import pyplot as plt
+import numpy as np
+from matplotlib.lines import Line2D
+import torch
```
Comment (OleguerCanal) [inline utilities/debug_utils.py]:
```suggestion
from matplotlib import pyplot as plt
from matplotlib.lines import Line2D
import numpy as np
import torch
```

File: utilities/debug_utils.py [hunk 2]
```diff
@@ -0,0 +1,126 @@
+import yaml
+import sys
+import os
```
Comment (OleguerCanal) [inline utilities/debug_utils.py]:
```suggestion
import os
import sys
import yaml
```

File: train.py [hunk 3]
```diff
@@ -17,30 +17,15 @@
...
+from utilities import debug_utils
 
 
-def train(params):
+def load_hgn(params, device, dtype):
     """Instantiate and train the Hamiltonian Generative Network.
 
     Args:
         params (dict): Experiment parameters (see experiment_params folder).
     """
```
Comment (OleguerCanal) [inline train.py:29]: Docstring

**Review by OleguerCanal — CHANGES_REQUESTED** [review]
